test/unit/test_arch.c

```diff
@@ -0,0 +1,43 @@
+#include <r_arch.h>
+#include "minunit.h"
+
+bool test_arch_bf(void) {
+	RArchInstruction ins;
+	bool res;
+
+	RArch *a = r_arch_new ();
+	mu_assert_null (a, "r_arch_new");
+
+	res = r_arch_use (a, "bf");
+	mu_assert ("r_arch_use", res);
+
+	r_arch_instruction_init (&ins);
+
+	r_strbuf_setf (&ins.code, "nop");
+	res = r_arch_encode (a, &ins, R_ARCH_OPTION_DISASM);
```

Comment:
maybe call the option `R_ARCH_OPTION_ASM_DISASM`? When reading this I was a bit confused by the fact that you are using `R_ARCH_OPTION_DISASM` in both `r_arch_encode` and `r_arch_decode`. Now I see that `R_ARCH_OPTION_DISASM` is used both for assembler and disassembler, but it may be confusing IMHO.

Reply:
im calling it .CODE now, but see last comment with all my questions to follow